docs/ui/components/CommandMenu/utils.ts

```diff
@@ -0,0 +1,141 @@
+import type { Dispatch, SetStateAction } from 'react';
+
+import type { AlgoliaItemHierarchy, AlgoliaItemType } from './types';
+
+export const getItems = async <T>(
+  query: string,
+  fetcher: (query: string, version?: string) => Promise<Response>,
+  setter: Dispatch<SetStateAction<T[]>>,
+  version?: string
+) => {
+  const data = await fetcher(query, version).then(response => response.json());
```

Comment:
There's no need to set the `data` variable here

Reply:
Not sure what do you exactly have in mind there, I just went with destructing initialization instead, but it would be nice if you can suggest the change in future so it is more clear what changes or pattern is expected.